modules/solid_mechanics/src/materials/CombinedPlasticityStressUpdate.C

```diff
@@ -0,0 +1,240 @@
+                                                     : 0),
+    _isotropic_hardening_constant(
+        this->isParamValid("isotropic_hardening_constant")
+            ? this->template getParam<Real>("isotropic_hardening_constant")
+            : 0),
+    _isotropic_hardening_function(this->isParamValid("isotropic_hardening_function")
+                                      ? &this->getFunction("isotropic_hardening_function")
+                                      : nullptr),
+    _yield_condition(-1.0), // set to a non-physical value to catch uninitalized yield condition
+    _isotropic_hardening_slope(0.0),
+    _plastic_strain(
+        this->template declareGenericProperty<RankTwoTensor, is_ad>(_base_name + "plastic_strain")),
+    _plastic_strain_old(
+        this->template getMaterialPropertyOld<RankTwoTensor>(_base_name + "plastic_strain")),
+    _kinematic_hardening_modulus(this->template getParam<Real>("kinematic_hardening_modulus")),
+    _gamma(this->template getParam<Real>("material_constant_gamma")),
+    Q(this->template getParam<Real>("Q")),
+    b(this->template getParam<Real>("b")),
+    _isotropic_hardening_variable(this->template declareGenericProperty<Real, is_ad>(
+        _base_name + "isotropic_hardening_variable")),
+    _isotropic_hardening_variable_old(
+        this->template getMaterialPropertyOld<Real>(_base_name + "isotropic_hardening_variable")),
+    _kinematic_hardening_variable(
+        this->template declareGenericProperty<Real, is_ad>("kinematic_hardening_variable")),
+    _kinematic_hardening_variable_old(
+        this->template getMaterialPropertyOld<Real>("kinematic_hardening_variable")),
+    _temperature(this->template coupledGenericValue<is_ad>("temperature"))
+{
+  if (parameters.isParamSetByUser("yield_stress") && _yield_stress <= 0.0)
+    mooseError("Yield stress must be greater than zero");
+  // Both of these parameters are given default values by derived classes, which makes them valid
+  if (_yield_stress_function == nullptr && !this->isParamValid("yield_stress"))
+    mooseError("Either yield_stress or yield_stress_function must be given");
+  if (!parameters.isParamValid("isotropic_hardening_constant") &&
+      !this->isParamValid("isotropic_hardening_function"))
+    mooseError(
+        "Either isotropic_hardening_constant or isotropic_hardening_function must be defined");
+  if (parameters.isParamSetByUser("isotropic_hardening_constant") &&
+      this->isParamValid("isotropic_hardening_function"))
+    mooseError(
+        "Only the isotropic_hardening_constant or only the isotropic_hardening_function can be "
+        "defined but not both");
+}
+
+template <bool is_ad>
+void
+CombinedPlasticityStressUpdateTempl<is_ad>::initQpStatefulProperties()
+{
+  _isotropic_hardening_variable[_qp] = 0.0;
+  _kinematic_hardening_variable[_qp] = 0.0;
+  _plastic_strain[_qp].zero();
+  RadialReturnStressUpdateTempl<is_ad>::initQpStatefulProperties();
```

Comment:
This needs to call the backstress implementation of this function.  
```suggestion
  BackstressRadialReturnStressUpdateTempl<is_ad>::initQpStatefulProperties();
```